File: P2A4_Herencias/C#/EmpleadoAsalariado.cs
using System;

public class EmpleadoAsalariado : Empleado
{
    private double salarioSemanal;

    public EmpleadoAsalariado(string nombre, string apellido, string nss, double salario)
        : base(nombre, apellido, nss)
    {
        SalarioSemanal = salario;
    }

    public double SalarioSemanal
    {
        get { return salarioSemanal; }
        set { salarioSemanal = (value >= 0.0) ? value : 0.0; }
    }

    public override double Ingresos()
    {
        return SalarioSemanal;
    }

    public override string ToString()
    {
        return $"empleado asalariado: {base.ToString()}\nsalario semanal: {SalarioSemanal:C}";
    }
}

File: P2A4_Herencias/Python/EmpleadoAsalariado.py
from Empleado import Empleado

class EmpleadoAsalariado(Empleado):
    def __init__(self, nombre, apellido, nss, salario):
        super().__init__(nombre, apellido, nss)
        self.salario_semanal = salario

    @property
    def salario_semanal(self):
        return self._salario_semanal

    @salario_semanal.setter
    def salario_semanal(self, salario):
        self._salario_semanal = salario if salario >= 0.0 else 0.0

    def ingresos(self):
        return self.salario_semanal

    def __str__(self):
        return f"empleado asalariado: {super().__str__()}\nsalario semanal: ${self.salario_semanal:,.2f}"
